sparse/src/KokkosSparse_BsrMatrix.hpp

```diff
@@ -625,25 +625,31 @@ class BsrMatrix {
     graph = staticcrsgraph_type(entries_device, row_map_device);
   }
 
-  /// \brief Constructor that accepts a row map, column indices, and
-  ///   values.
-  ///
-  /// The matrix will store and use the row map, indices, and values
-  /// directly (by view, not by deep copy).
-  ///
-  /// \param label
-  /// \param nrows [in] The number of rows.
-  /// \param ncols [in] The number of columns.
-  /// \param annz  [in] Filler for annz.
-  /// \param vals [in/out] The entries.
-  /// \param rows [in/out] The row map (containing the offsets to the
-  ///   data in each row).
-  /// \param cols [in/out] The column indices.
-  /// \param blockDimIn [in] The block dimensions.
-  BsrMatrix([[maybe_unused]] const std::string& label, const OrdinalType nrows,
-            const OrdinalType ncols, [[maybe_unused]] const size_type annz,
-            const values_type& vals, const row_map_type& rows,
-            const index_type& cols, const OrdinalType blockDimIn)
+/// \brief Constructor that accepts a row map, column indices, and
+///   values.
+///
+/// The matrix will store and use the row map, indices, and values
+/// directly (by view, not by deep copy).
+///
+/// \param label
+/// \param nrows [in] The number of rows.
+/// \param ncols [in] The number of columns.
+/// \param annz  [in] Filler for annz.
+/// \param vals [in/out] The entries.
+/// \param rows [in/out] The row map (containing the offsets to the
+///   data in each row).
+/// \param cols [in/out] The column indices.
+/// \param blockDimIn [in] The block dimensions.
+#if defined(DOXY)
```

Comment:
@e10harvey just to clarify the use of the guard here for the fix - apparently the attribute must proceed the argument for doxygen? And generally this should be supported, but due to compiler bug with gcc pre-9.3.0 (which does not properly support the use of attribute preceding the argument) instead for all cases the attribute follows the argument (which bypasses the compiler bug)? 
I had expected the guard to involve a check on the compiler and version, so I'm just trying to understand and untangle the connection with DOXY here

Reply:
My understanding is that for doxygen < 1.9.7, we need the attribute to be at the beginning. For GNU 8, we need the attribute to be at the end due to a compiler bug. The CI checks are installing the latest doxygen available in the CI VM, which is 1.9.6. Note that doxygen 1.9.7 has been released, though.